### PR TITLE
Add test for proper handling of connection failure to avoid 'cannot wait on socket event without a socket' error

### PR DESCRIPTION
**Pull request by knizhnik**
## Problem

See https://github.com/neondatabase/cloud/issues/14289
and PR #8210 

## Summary of changes

Add test for problems fixed in #8210

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Comment by kelvich** [comment]
I'll keep mentioning that we don't need https://github.com/neondatabase/neon/pull/8210/files#diff-d6b0f875ded76c09210876308df775e27adf4b424af1824c65a5d5871effe074L430-L434 now as `PQstatus` does the same check. Am I wrong about it or there is some reason to keep it?

**Comment by knizhnik** [comment]
> I'll keep mentioning that we don't need https://github.com/neondatabase/neon/pull/8210/files#diff-d6b0f875ded76c09210876308df775e27adf4b424af1824c65a5d5871effe074L430-L434 now as `PQstatus` does the same check. Am I wrong about it or there is some reason to keep it?

Sorry, I didn't notice your comment.
Yes, first check is redundant now.

**Comment by knizhnik** [comment]
> This does a full PS connect timeout up to an error; can we reduce that time so that we don't have a test consuming a runner's testslot for ~50 seconds?

Adding `statement_timeout` fixes this problem.

File: test_runner/regress/test_pageserver_reconnect.py [hunk 1]
```diff
@@ -40,3 +40,26 @@ def run_pgbench(connstr: str):
...
+    cur.execute("set statement_timeout='2s'")
+    cur.execute("SELECT setting FROM pg_settings WHERE name='neon.pageserver_connstring'")
+    connstring = cur.fetchall()[0][0]
+    cur.execute(
+        f"alter system set neon.pageserver_connstring='{connstring}?some_invalid_param=xyz'"
+    )
+    cur.execute("select pg_reload_conf()")
+    try:
+        cur.execute("select count(*) from pg_class")
+    except Exception as e:
```
Comment (jcsp) [inline test_runner/regress/test_pageserver_reconnect.py:63]: Can you make this exception handling more specific & match on the exception type + message that is expected?

Comment (knizhnik) [reply test_runner/regress/test_pageserver_reconnect.py:63]: Done

**Review by jcsp — APPROVED** [review]

**Pull request merged**